viewing_party/party.py

```diff
@@ -1,23 +1,138 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    new_movie = {}
+    if title == None or genre == None or rating == None:
+        return None
+    new_movie["title"] = title
+    new_movie["genre"] = genre
+    new_movie["rating"] = rating
+    return new_movie
 
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, title):
+    for ele in user_data["watchlist"]:  # ele is a dict
+        if ele["title"] == title:
+            user_data["watchlist"].remove(ele)
+            user_data["watched"].append(ele)
+    return user_data
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
 
+def get_watched_avg_rating(user_data):
+    if not user_data["watched"]:
+        return 0
+    else:
+        total = 0
+        for ele in user_data["watched"]:
+            total += ele["rating"]
+        return total/len(user_data["watched"])
+    # retun round( total/len(user_data["watched"]), 1)
+
+
+def get_most_watched_genre(user_data):
+    if user_data["watched"] == []:
+        return None

```

Comment:
We can make this guard clause more robust by checking if `user_data["watched"] or user_data`  is `Falsey`. In the rare case that `user_data` or `user_data["watched"]` is passed in as `None` or an empty set, list, or tuple then your code won't throw an error trying to find a non-existing "watched" key. 

```suggestion
    if not user_data["watched"]:
        return None

```